tools/pipelines/test-perf-benchmarks.yml

```
@@ -70,160 +85,311 @@ stages:
-                @ff-internal:registry=https://pkgs.dev.azure.com/fluidframework/internal/_packaging/build/npm/registry/
-                @microsoft:registry=https://office.pkgs.visualstudio.com/_packaging/Office/npm/registry/
-                always-auth=true
-              EOF
-              echo "Contents of .npmrc:"
+              echo Initialize package
+              npm init --yes
+
+              echo Generating .npmrc
+              echo "registry=https://registry.npmjs.org" >> ./.npmrc
+              echo "always-auth=false" >> ./.npmrc
+
+              echo "@fluidframework:registry=${{ variables.feed }}" >> ./.npmrc
+              echo "@fluid-experimental:registry=${{ variables.feed }}" >> ./.npmrc
+              echo "@fluid-internal:registry=${{ variables.feed }}" >> ./.npmrc
+              echo "@fluid-tools:registry=${{ variables.feed }}" >> ./.npmrc
+              echo "@ff-internal:registry=https://pkgs.dev.azure.com/fluidframework/internal/_packaging/build/npm/registry/" >> ./.npmrc
+              echo "@microsoft:registry=https://office.pkgs.visualstudio.com/_packaging/Office/npm/registry/" >> ./.npmrc
+              echo "always-auth=true" >> ./.npmrc
               cat .npmrc
 
+        # Auth to internal feed
         - task: npmAuthenticate@0
           displayName: 'npm authenticate (internal feed)'
           inputs:
-            workingFile: $(toolAbsolutePath)/.npmrc
+            workingFile: ${{ variables.testWorkspace }}/.npmrc
 
+        # Auth to Office feed
         - task: npmAuthenticate@0
           displayName: 'npm authenticate (office feed)'
           inputs:
-            workingFile: $(toolAbsolutePath)/.npmrc
+            workingFile: ${{ variables.testWorkspace }}/.npmrc
             customEndpoint: 'Office Packages - Readonly'
 
-        - task: Npm@1
-          displayName: 'Install @ff-internal/aria-logger'
+        - task: Bash@3
+          displayName: Print Parameters
           inputs:
-            command: 'custom'
-            workingDir: $(toolAbsolutePath)
-            customCommand: 'install @ff-internal/aria-logger'
-            customRegistry: 'useNpmrc'
+            targetType: 'inline'
+            script: |
+              echo "
+              Pipeline Parameters:
+                poolBuild=${{ parameters.poolBuild }}
+                Build=${{ parameters.taskBuild }}
+
+              Task Variables:
+                releaseBuild=$(releaseBuildVar)
+                toolAbsolutePath=$(toolAbsolutePath)
+                artifactPipeline=$(artifactPipeline)
+                feed=$(feed)
+                BuildDir=${{ variables.buildDirectory }}
+                TestWorkspace=${{ variables.testWorkspace }}
 
-        - task: Npm@1
-          displayName: 'Run performance test - execution time'
+              Build Params
+                SourceBranch=$(Build.SourceBranch)
+              "
+
+        - task: Bash@3
+          displayName: 'Install aria-logger'
```

```suggestion
          displayName: 'Prepare telemetry reporter'
```

Does it feel like a more accurate description of this step? Installing aria logger is part of it, yes, but overall what we're achieving here is setting up telemetry-logger so it can be run later in the pipeline.